RecoLocalTracker/SiPixelRecHits/src/SiPixelTemplate.cc

```diff
@@ -133,36 +133,24 @@ bool SiPixelTemplate::pushfile(int filenum, std::vector< SiPixelTemplateStore >
    // Local variables
    int i, j, k, l;
    float qavg_avg;
-   const char *tempfile;
    //	char title[80]; remove this
    char c;
    const int code_version={17};
-   
-   
-   
    //  Create a filename for this run
-   
-   std::ostringstream tout;
-   
    //  Create different path in CMSSW than standalone
-   
+   std::string tempfile=std::to_string(filenum);
+   for (unsigned cnt=4-tempfile.length(); cnt > 0; cnt-- ){ tempfile ="0"+tempfile; }
+   tempfile = "template_summary_zp" + tempfile + ".out";
+
 #ifndef SI_PIXEL_TEMPLATE_STANDALONE
-   tout << "CalibTracker/SiPixelESProducers/data/template_summary_zp"
-   << std::setw(4) << std::setfill('0') << std::right << filenum << ".out" << std::ends;
-   std::string tempf = tout.str();
-   edm::FileInPath file( tempf.c_str() );
+   tempfile = "CalibTracker/SiPixelESProducers/data/" + tempfile; 
+   edm::FileInPath file( tempfile.c_str() );
    tempfile = (file.fullPath()).c_str();
```

Comment:
@smuzaffar @mrodozov @perrotta 
I'm confused by this line.  tempfile was originally a char array.   This was deleted and now tempfile is a std::string.  So what is going to happen in this assignment?